### ISSUE TITLE
[HOLD for payment 2023-05-29] [HOLD for payment 2023-05-25] Prevent creating new requests from within a settled expense report

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#19028]
@trjExpensify @JmillsExpensify @luacmartins @mountiny 

### Problem

Currently, the requester can continue to create requests from within a settled expense report. _This is not possible for the payee after refreshing the page._ 

**Expected**
![Screenshot 2023-05-16 at 17 01 09](https://github.com/Expensify/App/assets/10736861/74b9679b-ccae-4297-aa4e-e3f6cd1d8067)

**Current**
![Screenshot 2023-05-16 at 17 01 39](https://github.com/Expensify/App/assets/10736861/c9e4189a-76ae-4401-9ed2-23fc7cf119d0)
![Screenshot 2023-05-16 at 17 01 46](https://github.com/Expensify/App/assets/10736861/ad9b1f46-7f86-4637-9cd4-397eca957148)

### Solution

Blocks users from:

- requesting money from within a settled IOUReport or expense report
- or sending money from an iouReport in any state
- sending money from expense report in any state
- requesting or sending money from transaction thread

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01be7b10bebf499be1</li>
        <li>Upwork Job ID: 1658503343790088192</li>
        <li>Last Price Increase: 2023-05-16</li>
    </ul>
</details>

**Comment by trjExpensify** [comment]
Nice, I agree you shouldn't be able to `request money` within a settled report. You shouldn't be able to `Send money` within any expense/iouReport. 

@thienlnam, should you be able to `Assign tasks` within an expense/iouReport? Seems like that might break. Outside of the scope of this PR though!

**Comment by thienlnam** [comment]
> @thienlnam, should you be able to Assign tasks within an expense/iouReport? Seems like that might break. Outside of the scope of this PR though!

Actually, you should be able to assign tasks anywhere - I'm not sure if it works out of the box right now because that hasn't been our focus for V1 - but technically yes should be possible!

**Comment by trjExpensify** [comment]
Yeah, is that going to break though? In which case, do we want to not show the `Assign Task` option until we firm up on how it's possible to add a task to an expense/iouReport?

**Comment by thienlnam** [comment]
It actually does work out of the box - there is a bug with the avatar in the LHN but we can fix that along with the the avatars for requests

**Comment by trjExpensify** [comment]
Cool. I _think_ avatars for requests is being fixed if you look at the latest we're working on from here: https://expensify.slack.com/archives/C03TQ48KC/p1684276084051879?thread_ts=1684266162.911009&cid=C03TQ48KC

**Comment by luacmartins** [comment]
No longer on hold

**Comment by mountiny** [comment]
Taking this over also updated the solution with more cases we will block

**Comment by mountiny** [comment]
PR merged, @rushatgabhane $1000 flat rate for the PR help

**Comment by trjExpensify** [comment]
Errr.. why are there two payment dates on this issue? 😕

**Comment by Julesssss** [comment]
I've seen this on quite a few issues over the last 2 weeks.

**Comment by trjExpensify** [comment]
So are we waiting on the 29th or is this due today?

**Comment by trjExpensify** [comment]
In the meantime, offer sent to Satish in prep @rushatgabhane 👍

**Comment by trjExpensify** [comment]
[PR hit prod](https://github.com/Expensify/App/pull/19033#issuecomment-1552567919) 7 days ago. Paid and closing.

**Comment by Julesssss** [comment]
> So are we waiting on the 29th or is this due today?

Yeah, the first date was correct.
